ui/components/component-library/button/button.stories.js

```diff
@@ -0,0 +1,213 @@
+import React from 'react';
+import {
+  ALIGN_ITEMS,
+  DISPLAY,
+  FLEX_DIRECTION,
+  SIZES,
+  TEXT,
+} from '../../../helpers/constants/design-system';
+import { ICON_NAMES } from '../icon';
+import { BUTTON_LINK_SIZES } from '../button-link/button-link.constants';
+import Box from '../../ui/box/box';
+import { Text } from '../text';
+import README from './README.mdx';
+import { Button, BUTTON_TYPES } from '.';
+
+const marginSizeControlOptions = [
+  undefined,
+  0,
+  1,
+  2,
+  3,
+  4,
+  5,
+  6,
+  7,
+  8,
+  9,
+  10,
+  11,
+  12,
+  'auto',
+];
+
+export default {
+  title: 'Components/ComponentLibrary/Button',
+  id: __filename,
+  component: Button,
+  parameters: {
+    docs: {
+      page: README,
+    },
+    controls: { sort: 'alpha' },
+  },
+  argTypes: {
+    as: {
+      control: 'select',
+      options: ['button', 'a'],
+    },
+    block: {
+      control: 'boolean',
+    },
+    children: {
+      control: 'text',
+    },
+    className: {
+      control: 'text',
+    },
+    danger: {
+      control: 'boolean',
+    },
+    disabled: {
+      control: 'boolean',
+    },
+    href: {
+      control: 'text',
+    },
+    icon: {
+      control: 'select',
+      options: Object.values(ICON_NAMES),
+    },
+    iconPositionRight: {
+      control: 'boolean',
+    },
+    iconProps: {
+      control: 'object',
+    },
+
+    loading: {
+      control: 'boolean',
+    },
+    size: {
+      control: 'select',
+      options: Object.values(BUTTON_LINK_SIZES),
+    },
+    type: {
+      options: Object.values(BUTTON_TYPES),
+      control: 'select',
+    },
+    marginTop: {
+      options: marginSizeControlOptions,
+      control: 'select',
+      table: { category: 'box props' },
+    },
+    marginRight: {
+      options: marginSizeControlOptions,
+      control: 'select',
+      table: { category: 'box props' },
+    },
+    marginBottom: {
+      options: marginSizeControlOptions,
+      control: 'select',
+      table: { category: 'box props' },
+    },
+    marginLeft: {
+      options: marginSizeControlOptions,
+      control: 'select',
+      table: { category: 'box props' },
+    },
+  },
+  args: {
+    children: 'Button',
+  },
+};
+
+export const DefaultStory = (args) => <Button {...args} />;
+
+DefaultStory.storyName = 'Default';
+
+export const Type = (args) => (
+  <Box display={DISPLAY.FLEX} gap={1}>
+    <Button type={BUTTON_TYPES.PRIMARY} {...args}>
+      Primary Button
+    </Button>
+    <Button type={BUTTON_TYPES.SECONDARY} {...args}>
+      Secondary Button
+    </Button>
+    <Button type={BUTTON_TYPES.LINK} {...args}>
+      Link Button
```

Comment:
Let's update this text so it's consistent with our component naming
```suggestion
     Button Primary
    </Button>
    <Button type={BUTTON_TYPES.SECONDARY} {...args}>
      Button Secondary
    </Button>
    <Button type={BUTTON_TYPES.LINK} {...args}>
     Button Link
```